### PR TITLE
Add support for `<style>` tags

### PR DESCRIPTION
**Pull request by rafaeltonholo**
## Summary by CodeRabbit

- **New Features**
	- Introduced CSS parsing capability within the SVG parser.
	- Added support for `ComputedRule` to enhance CSS specificity handling.
	- Enhanced SVG element processing to include style nodes.
	- Updated `asNode` methods across various SVG node classes to accept computed rules.

- **Bug Fixes**
	- Improved error handling for clip paths and gradient processing in SVG nodes.

- **Tests**
	- Comprehensive unit tests added for CSS parsing and SVG handling, covering various edge cases and ensuring robustness.

- **Documentation**
	- Enhanced documentation for new methods and classes related to CSS parsing and SVG style handling.

### CODE REVIEW

File: svg-to-compose/src/commonMain/kotlin/dev/tonholo/s2c/lexer/css/token/consumer/HashTokenConsumer.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return if (next != Char.EMPTY && next.isHexDigit()) {
+            var backwardLookupOffset = 0
+            do {
+                val prev = iterator.peek(--backwardLookupOffset)
+                if (prev == Char.EMPTY || prev != ' ') {
+                    break
+                }
+            } while (prev == ' ')
+
+            if (iterator.peek(backwardLookupOffset) in CssTokenKind.Colon) {
```
[the remark this answers is not in the document]
Comment (rafaeltonholo) [reply svg-to-compose/src/commonMain/kotlin/dev/tonholo/s2c/lexer/css/token/consumer/HashTokenConsumer.kt:26]: The `in` uses the operator fun `contains`. The `TokenKind` interface has a default implementation of the operator `contains` that receives a `Char`. As the `CssTokenKind` implements that interface, the current code is correct.
